server/src/test/java/org/elasticsearch/index/engine/EngineConfigTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.engine;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.IndexSettingsModule;
+
+
+public class EngineConfigTests extends ESTestCase {
```

Comment:
I do not think we need a new class for this. Maybe there is an existing class where this would be appropriate?

Reply:
Thanks @jasontedor , how about elasticsearch/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java or elasticsearch/server/src/test/java/org/elasticsearch/index/engine/InternalEngineSettingsTests.java ?

Reply:
I moved the test to ```InternalEngineTests.java```. I'm not sure if it's appropriate. Please review again, thanks!